File: java/Payment/src/main/java/com/maicol/payment/Discounts.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.maicol.payment;

/**
 *
 * @author Maicol
 */

public class Discounts {
    private String errorMessage = "";
    private double arl = 0;
    private double pension = 0;
    private double health = 0;
    private double totalDiscounts = 0;

    public double getArl(int risk) {
        switch(risk) {
            case 1 -> { return 0.00522; }
            case 2 -> { return 0.01044; }
            case 3 -> { return 0.02436; }
            case 4 -> { return 0.04350; }
            default -> {
                this.errorMessage = "This level doesn't exist";
                return 0;
            }
        }
    }

    public void initDiscounts(int salary, int risk) {
        this.arl = salary * getArl(risk);
        this.pension = salary * 0.04;
        this.health = salary * 0.04;
        this.totalDiscounts = this.health + this.pension + this.arl;
    }

    public void showDiscounts() {
        System.out.println("Pension: " + this.pension);
        System.out.println("Health: " + this.health);
        System.out.println("Arl: " + this.arl);
        System.out.println("Total discount: " + this.totalDiscounts);
    }

    public String getErrorMessage() {
        return this.errorMessage;
    }

    public double getTotalDiscounts() {
        return this.totalDiscounts;
    }
}

File: php/exercises/1.payment/php/discounts.php
<?php
trait discounts{
    private string $errorMensage="";
    private float $arl=0;
    private float $pension=0;
    private float $health=0;
    private float $totalDiscounts=0;
    
    public function getArl(int $risk):float{
        if($risk==1){return 0.00522;
        }
        else if($risk==2){return 0.01044;
        }
        else if($risk==3){return 0.02436;
        }
        else if($risk==4){return 0.04350;
        }    
        else{$this->errorMensage=" This level no exists";
            return 0;
        }
    }
    public function initDiscounts(int $salary,int $risk){
        $this->arl=$salary*$this->getArl($risk);
        $this->pension=$salary*0.04;
        $this->health=$salary*0.04;
        $this->totalDiscounts=$this->health+$this->pension+$this->arl;
    }
    public function showDiscounts():void{
        echo "<p>Pension: ". $this->pension."</p>";
        echo "<p>Health: ".$this->health."</p>";
        echo "<p>Arl: ".$this->arl."</p>";
        echo "<p>Total discount: ".$this->totalDiscounts."</p>";
    }
}
?>